src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(url, options)
+      .then(this._checkResponse)
+      .then((res) => res.json())
```

Comment:
`.then(this._checkResponse)` is not needed
`.then((res) => res.json())` is not needed

You have `renderResult` for checking the response

```js

  _request(url, options) {
    return fetch(url, options)
      .then(this.renderResult)
  }
```